src/frontend/src/optimizer/plan_node/stream_hash_join.rs

```diff
@@ -305,9 +440,7 @@ impl fmt::Display for StreamHashJoin {
             && self.right().watermark_columns().contains(rjk)
         {
             f.debug_struct("StreamWindowJoin")
-        } else if self.clean_left_state_conjunction_idx.is_some()
-            && self.clean_right_state_conjunction_idx.is_some()
-        {
+        } else if self.band_condition.is_some() {
```

Comment:
> From the user's perspective, `StreamIntervalJoin` means it can utilize the input's watermark by band conditions. If there are no watermarks, no difference for users, so let's change it back to `StreamHashJoin`. `BandJoin` is just an optimization for our implementation.

fixed.